File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends


class Solution
{
    public:
    //Function to reverse words in a given string.
    string reverseWords(string S) 
    { 
        // code here 
        vector<string> rev;
        string temp="";
        for(auto a:S){
            if(a=='.'){
                rev.emplace_back(temp);
                rev.emplace_back(".");
                temp.clear();
            }
            else
                temp+=a;
        }
        
        for(int i=rev.size()-1;i>=0;--i){
            temp+=rev[i];
        }
        return temp;
    } 
};

// { Driver Code Starts.
int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        Solution obj;
        cout<<obj.reverseWords(s)<<endl;
    }
}  // } Driver Code Ends

File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.py

# User function Template for python3

class Solution:
    
    #Function to reverse words in a given string.
    def reverseWords(self,s):
        # code here 
        s=s.split('.')
        s=s[::-1]
        return ".".join(s)

#{ 
#  Driver Code Starts
# Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        s = str(input())
        obj = Solution()
        print(obj.reverseWords(s))

# } Driver Code Ends